### PR TITLE
More test fixes and related bug fixes

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Make PriorityProducerConsumer clear its queues on dispose
- Raise AnalysisQueue.AnalysisComplete at the end of consumer loop
- Derive IModule from IAnalysisValue
- Add more assertions
- Fix more tests

### CODE REVIEW

File: src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs [hunk 1]
```diff
@@ -58,7 +58,17 @@ internal sealed class PriorityProducerConsumer<T> : IDisposable {
             _comparer = comparer ?? EqualityComparer<T>.Default;
         }
 
-        public void Dispose() => _disposeToken.TryMarkDisposed();
+        public void Dispose() {
+            if (_disposeToken.TryMarkDisposed()) {
+                lock (_syncObj) {
+                    Debug.Assert(_pendingTasks.Count == 0 || _firstAvailablePriority == _maxPriority);
+                    _pendingTasks.Clear();
```
Comment (MikhailArkhipov) [inline src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs]: Should we cancel `pendingTcs`?

File: src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs [hunk 2]
```diff
@@ -58,7 +58,17 @@ internal sealed class PriorityProducerConsumer<T> : IDisposable {
...
+                    foreach (var queue in _queues) {
+                        queue.Clear();
+                    }
+                }
+            };
+        }
 
         public void Produce(T value, int priority = 0) {
             if (priority < 0 || priority >= _maxPriority) throw new ArgumentOutOfRangeException(nameof(priority));
```
Comment (MikhailArkhipov) [inline src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs]: If you want, you can use `Check` that is now in infra

File: src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs]: They are canceled by `_disposeToken` itself:
https://github.com/Microsoft/python-language-server/blob/1d0c6244463ba740896ca78cc444f334b97454c0/src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs#L135

File: src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs (hunk 2 above)
Comment (AlexanderSher) [reply src/Analysis/Engine/Impl/Infrastructure/PriorityProducerConsumer.cs]: Good to know!

**Pull request merged**